glue_astronomy/translators/nddata.py

```diff
@@ -190,6 +190,8 @@ def to_object(self, data_or_subset, attribute=None):
         else:
             data = data_or_subset
 
+        if attribute is None and "flux" in data.main_components:
+            attribute = "flux"
```

Comment:
```suggestion
        if attribute is None and "flux" in data.main_components:
            attribute = "flux"
```
```suggestion
        if attribute is None and len(data.main_components) > 1 and "flux" in data.main_components:
            attribute = "flux"
```
Otherwise this would change behaviour in the one case of a single component that is named differently (though that's obviously not tested, so may be not really relevant). It might simplify things to handle this all inside `_get_attribute` at this point, though it should get some documentation here, e.g.
```python
        attribute : `glue.core.component_id.ComponentID`
            The attribute to use for the StdDevUncertainty data. If not specified,
            will use a single component or try to identify one named "flux".
```